go/enclave/enclave.go

```diff
@@ -251,27 +251,28 @@ func (e *enclaveImpl) Status() (common.Status, error) {
 		if errors.Is(err, errutil.ErrNotFound) {
 			return common.AwaitingSecret, nil
 		}
-		return common.Unavailable, err
+		return common.Unavailable, e.internalError(err)
 	}
 	return common.Running, nil // The enclave is local so it is always ready
 }
 
 // StopClient is only implemented by the RPC wrapper
-func (e *enclaveImpl) StopClient() error {
+func (e *enclaveImpl) StopClient() common.SystemError {
 	return nil // The enclave is local so there is no client to stop
 }
 
 // SubmitL1Block is used to update the enclave with an additional L1 block.
-func (e *enclaveImpl) SubmitL1Block(block types.Block, receipts types.Receipts, isLatest bool) (*common.BlockSubmissionResponse, error) {
+func (e *enclaveImpl) SubmitL1Block(block types.Block, receipts types.Receipts, isLatest bool) (*common.BlockSubmissionResponse, common.SystemError) {
 	if atomic.LoadInt32(e.stopInterrupt) == 1 {
-		return nil, nil //nolint:nilnil
+		return nil, e.internalError(fmt.Errorf("requested SubmitL1Block with the enclave stopping"))
```

Comment:
Good spot - I went searching for inconsistencies and found the following:

- `verifyAttestationAndEncryptSecret` does this if check but doesn't really make a lot of sense because the upstream call of it is `SubmitL1Block` which already does the check. Removed.
- `Status()` is also inconsistent, I thought that was on purpose on my first pass, but I don't see why it would be after looking further into it. Changed to the same pattern.